src/main.rs

```diff
@@ -127,13 +130,28 @@ fn prune(dir: ReadDir) {
         }
 
         println!("    Pruning for lib {:?}", lib);
-        let _ = lib_paths.pop();
+        let latest = unwrap!(unwrap!(unwrap!(lib_paths.pop()).metadata()).modified());
```

Comment:
Why do we pop here? I thought we wanted to delete all the files or none to avoid step lock delete?
i.e first go through the element and check if any is outside the range (maybe using map and any), then go through the element and remove them all.

Reply:
I didn't get you, so what is the step locking problem in this ? We arrange chronologically ascending, so the last one is the latest and if any have time stamps that differ by more than 2 hrs wrt the latest we delete them. Can you say what problem does this create ?

Reply:
OK discussed with @jeanphilippeD - so what will happen is say there are 4 dups and only one got updated and is latest by more than 2hrs - so we will delete the prev 4 and keep this one (A). Next time all the others (3 of them) are going to be built and this A is going to be delete. Next time A is going to be built again and the other 3 are going to be deleted and this continues. So it's not going to recover. So we delete all if > 2hrs time diff is found.

Reply:
File A1 and A2 are duplicate.
Version of A1 changed:

Build/prune 1:
Build A1
A1 is most rencent
Keep A1 delete A2

No change, Build/prune 2:
Build missing A2
A2 is most rencent
Keep A2 delete A1

Build 3 -> delete A1

To avoid this issue we need to delete all:


Build/prune 1:
Build A1
A1 is most rencent
delete A1, A2

No change, Build/prune 2:
Build missing A1, A2
A1 A2 have similar date: keep both